src/ServiceControl.Transports.AmazonSQS/ServiceControlSqsTransport.cs

```diff
@@ -0,0 +1,47 @@
+namespace ServiceControl.Transports.AmazonSQS
+{
+    using System;
+    using System.Data.Common;
+    using System.Linq;
+    using Amazon;
+    using NServiceBus;
+    using NServiceBus.Settings;
+    using NServiceBus.Transport;
+
+    public class ServiceControlSqsTransport : SqsTransport
+    {
+        public override TransportInfrastructure Initialize(SettingsHolder settings, string connectionString)
+        {
+            var builder = new DbConnectionStringBuilder { ConnectionString = connectionString };
+
+            PromoteEnvironmentVariableFromConnectionString(builder, "AccessKeyId", "AWS_ACCESS_KEY_ID");
+            PromoteEnvironmentVariableFromConnectionString(builder, "SecretAccessKey", "AWS_SECRET_ACCESS_KEY");
+            var region = PromoteEnvironmentVariableFromConnectionString(builder, "Region", "AWS_REGION");
+
+            var awsRegion = RegionEndpoint.EnumerableAllRegions
+                .SingleOrDefault(x => x.SystemName == region);
+
+            if (awsRegion == null)
+            {
+                throw new ArgumentException($"Unknown region: \"{region}\"");
+            }
+
+            settings.Set("NServiceBus.AmazonSQS.Region", awsRegion);
+
+            // SQS doesn't support connection strings so pass in null.
+            return base.Initialize(settings, null);
+        }
+
+        static string PromoteEnvironmentVariableFromConnectionString(DbConnectionStringBuilder builder, string connectionStringKey, string environmentVariableName)
+        {
+            if (builder.TryGetValue(connectionStringKey, out var value))
+            {
+                var valueAsString = (string) value;
+                Environment.SetEnvironmentVariable(environmentVariableName, valueAsString, EnvironmentVariableTarget.Process);
```

Comment:
Are there special permissions required to do this? If we run the SC windows service as a network user or local machine will it be able to update the process environment variables? I think it should be OK, just checking.

Reply:
Not thatI know. Will check but the assumption was a process has always right to write env stuff for itself

Reply:
@mikeminutillo couldn't find anything mentioning the process cannot set its own variables. The good thing about this approach is that the env variables are bound to the lifetime of the process, if the process goes the env variables are gone as well.